pkg/apis/core/types.go

```diff
@@ -1934,6 +1934,9 @@ type Probe struct {
 	// Minimum consecutive failures for the probe to be considered failed after having succeeded.
 	// +optional
 	FailureThreshold int32
+	// The list of a container(s) which is(are) restarted when Liveness probe detects a failure of the monitored resource.
+	// +optional
+	MonitoredContainers []string
```

Comment:
copying in orphaned conversation from https://github.com/kubernetes/kubernetes/commit/6e1f1315c181d30f1eced895be0d0c6e3a031344#commitcomment-28187988

@liggitt:
> This makes for odd semantics if the container that defines this liveness check isn't in the list of monitored containers. If we wanted the ability to exec a liveness check in a container other than the one defining the liveness probe, a containerName field in the ExecAction struct would make more sense to me

@liggitt:
> For kubelets that don't know about this field, this would also fail in odd ways (when the check fails, this container would be restarted or fail, not the monitoredContainers)

@sbezverk:
> @liggitt Thank you for your review. IMHO having list of containers which are needed to be restarted as a part of a liveness side car provides more flexibility, for the case when it is required to restart more than one container in case of a main container failure.
> With regards to your suggestion of adding container name to ExecAction I fail to see its value. It looks more difficult to parse and it does not have added flexibility or restarting multiple containers. I would appreciate if you explain a bit why it makes more sense to you.

@sbezverk:
> This attribute should be completely optional. If it is empty then the behaviour would be exactly as it was before, kubelet will restart the container running liveness probe. If this list is populated then kubelet should restart containers in the list and then restart the container running liveness prob.

@liggitt:
> The value is that it lets you accomplish your goal of running liveness checks using a container that has tools/commands that are not in the container you want to check liveness of.
> 
> Some of the reasons I prefer modifying ExecAction to specify the container to use to run the exec:
> 1. It keeps the liveness check connected to the container it applies to. If you remove that container, the associated liveness check goes away as well.
> 2. It puts the information (which container to use to perform the exec) exactly where it is required, with minimal disruption to any other logic in the kubelet.
> 3. It maintains the 1:1 relationship between containers and liveness checks. Changing that to a many-to-one relationship as this PR proposes means that a liveness check failure can ripple into n container terminations, which seems like a significant change
> 4. It degrades better on kubelets that are unaware of the new field (they either run the exec check in the container itself or fail entirely, rather than try to kill the wrong container on failure)

@sbezverk:
> I guess my confusion was that I was not aware that it is possible to run in the container a binary located in one another container. If you could point me to some examples, that would be great. thank so much.

@liggitt:
>> I was not aware that it is possible to run in the container a binary located in one another container
>
> I don't think it is. That's not what I meant to imply.
> 
> Isn't the goal to exec a command in container B that indicates whether container A is alive? I'd still expect that check to be defined as part of container A (it is defining how to check for liveness of A). The fact that it runs in container B is just an implementation detail of the particular probe method.
> 
> Come to think of it, if the tools you need to perform a liveness check aren't available in the main container, I think you can already accomplish a liveness check via a sidecar container using a httpGet action